service/prometheus/update.go

```diff
@@ -0,0 +1,41 @@
+package prometheus
+
+import (
+	"github.com/prometheus/prometheus/config"
+)
+
+// isManaged returns true if the given scrape config is managed by the prometheus-config-controller,
+// false otherwise.
+func isManaged(scrapeConfig config.ScrapeConfig) bool {
+	for _, targetGroup := range scrapeConfig.ServiceDiscoveryConfig.StaticConfigs {
+		if _, ok := targetGroup.Labels[ClusterLabel]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+// UpdateConfig takes an existing Prometheus configuration,
+// and a list of Prometheus scrape configurations.
+// A new configuration is returned, that includes both the scrape configurations
+// in the prior configuration, as well as the new scrape configs.
```

Comment:
What is the strategy when you align comments? I never see any rule and this makes my heart cry deep inside. I wake up in the night and scream "WHYYYY?".

Reply:
I tend to break by clauses, but no rough ruling. Going with what reads well.